### PR TITLE
Refactored: Disable datadog only if DISABLE_DATADOG_RAILS is set but not '0'

### PR DESCRIPTION
**Pull request by fruwe**
I would like to be able to have a default value for DISABLE_DATADOG_RAILS in order to disable datadog by default. I only want to enable datadog, if the value is something like 0. Unfortunately, my environment requires the DISABLE_DATADOG_RAILS always to be set to something, otherwise it will default to the default value. Since the current code only checks for the existence of the environment value, any value, even an empty string will disable datadog, thus making it impossible to enable datadog without removing the variable altogether.

With this PR I am proposing to disable datadog with DISABLE_DATADOG_RAILS only if its value is not '0'

### CODE REVIEW

File: lib/ddtrace/contrib/rails/patcher.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def patched?
           end
 
           def compatible?
-            return if ENV['DISABLE_DATADOG_RAILS']
+            return if ENV['DISABLE_DATADOG_RAILS']&.match(/^(true|t|yes|y|1|disable|disabled)?$/i)
```
Comment (vochicong) [inline lib/ddtrace/contrib/rails/patcher.rb]: Isn't it confusing if DISABLE_DATADOG_RAILS=disable or disabled ?

Comment (fruwe) [reply lib/ddtrace/contrib/rails/patcher.rb]: Oh, yes, you are right. `true` should not be `disabled`.

I guess, ENABLE_DATADOG_RAILS would have been less confusing than double negation...

How about this:
`return if ENV['DISABLE_DATADOG_RAILS'] && ENV['DISABLE_DATADOG_RAILS'] != '0'`

**Comment by vochicong** [comment]
https://circleci.com/gh/DataDog/dd-trace-rb/2808?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
Testing on CircleCI failed with Ruby 2.2.7. It looks like that `&.` didn't work.

File: lib/ddtrace/contrib/rails/patcher.rb (hunk 1 above)
Comment (vochicong) [reply lib/ddtrace/contrib/rails/patcher.rb]: `return if ENV['DISABLE_DATADOG_RAILS'] && ENV['DISABLE_DATADOG_RAILS'] != '0'` looks great 👍

**Comment by fruwe** [comment]
Thanks. I changed it to the simplified version

**Comment by fruwe** [comment]
@delner Thank you very much; I changed the PR accordingly.

If you require more values, then something like this
`return unless ENV['DISABLE_DATADOG_RAILS'].to_s.match(/^(0|f|false)?$/i)`
might also work (and would allow capital letters as well).

However, your suggestion makes it easier to understand on what values disable the disable flag

**Comment by delner** [comment]
@fruwe True, I like your suggestion as well.

**Comment by delner** [comment]
I know this change is so very small, but do you think we could add a test for this? If it isn't too difficult?

**Comment by fruwe** [comment]
sure, I will do this later

**Comment by delner** [comment]
@fruwe Just a heads up, we're aiming for a release next week; I'd love to get this in before then, if at all possible!

**Comment by delner** [comment]
@fruwe We're planning on releasing `0.13.0` Wednesday June 20th, so it'd be good to get this in. Otherwise I think it'll have to wait until `0.14.0`.

**Comment by delner** [comment]
Looks like this PR has become a bit stale. I think I'm going to close this for now. If you'd like to re-open it, please feel free to comment here again and we can reconsider it.

**Pull request closed without merging**